pkg/provisioner/crdprovisioner.go

```diff
@@ -627,61 +628,74 @@ func (c *CrdProvisioner) UnpublishVolume(
 		return err
 	}
 
-	// if volume's maxMountReplicaCount is 0, is a block volume, or attachment is a replica, detach
-	if azVolumeInstance.Spec.MaxMountReplicaCount == 0 || azVolumeAttachmentInstance.Spec.RequestedRole == diskv1beta1.ReplicaRole {
+	// if volume's maxMountReplicaCount is 0, is a block volume, detach
+	if azVolumeInstance.Spec.MaxMountReplicaCount == 0 {
 		// if volume's maxMountReplicaCount == 0, detach and wait for detachment to complete
-		// if AzVolumeAttachment instance indicates that previous attachment request was successful, annotate the CRI with detach request so that the underlying volume attachment can be properly detached.
-		if azVolumeAttachmentInstance.Status.Detail != nil {
-			// if detachment is pending, return to prevent duplicate request
-			if azVolumeAttachmentInstance.Status.State == diskv1beta1.Detaching {
-				return status.Errorf(codes.Aborted, "detachment still in process for volume (%s) and node (%s)", volumeName, nodeID)
-			}
+		return c.detachVolume(ctx, azVAClient, azVolumeAttachmentInstance)
+	}
 
```

Comment:
I think it would be more consistent to refactor this into a separate `demote` function.